### PR TITLE
[FIX] website, *: not fail on visitors views without event rights

### PR DESCRIPTION
**Pull request by bso-odoo**
*: website_crm, website_event, website_event_track

Before this commit, when accessing the visitor views without having any
event rights, an error was triggered.

This commit restricts the display of the `event_registration_count`
field on the `website_visitor_view_form` restricted to the
`event.group_event_user` group so that its invisible condition does not
fail. Similarly it restricts a visibility condition on
`event_track_wishlisted_count` to the `event.group_event_manager` group.
To avoid encountering the same issue with further extensions of
`_compute_email_phone` on the `website.visitor` model, we make its
fields `compute_sudo=True`, and revert the specific `sudo` that had
been added in [1].

Steps to reproduce:
- install `website_event`
- set Demo user to have no Events right, but be an "Editor and Designer"
for Website
- register to an event as an anonymous user
- access the Visitors page as Demo user

=> An error was produced.

Steps to reproduce (2):
- install `website_event_track`
- set Demo user to have no Events right, but be an "Editor and Designer"
- access the Visitors page as Demo user

=> An error was produced.

[1]: https://github.com/odoo/odoo/commit/5ba09c2cb2bb180cfb1363c352f35bd8cf3b132b

opw-3537849

### CODE REVIEW

File: addons/website_event/models/website_visitor.py
Comment (rdeodoo) [inline addons/website_event/models/website_visitor.py:46]: As it's yet another `sudo()` in `_compute_email_phone()`, don't you think it would be better to flag that compute method as `compute_sudo=True`?

**Review by rdeodoo — APPROVED** [review]
I'll wait for your feedback before merging.
AFAIC it can be shipped as-is but I wonder if we shouldn't fix the issue at a lower level once and for all.

File: addons/website_event/models/website_visitor.py
Comment (bso-odoo) [reply addons/website_event/models/website_visitor.py:46]: I was not sure we could make such a change to a field in stable. If this is ok then I'll do it (and remove the other added `sudo`s).

**Comment by bso-odoo** [comment]
@rdeodoo Changed to `compute_sudo`, reverted old PR's `sudo` and adapted description accordingly.

File: addons/website_event/models/website_visitor.py
Comment (rdeodoo) [reply addons/website_event/models/website_visitor.py]: I'am 99% sure it's ok. I asked confirmation just to be sure before r+.
Ty for the change

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

File: addons/website_event/views/website_visitor_views.xml
Comment (rdeodoo) [inline addons/website_event/views/website_visitor_views.xml:38]: I can't click on any visitor (even without registration), could you have a look?

**Review by rdeodoo — CHANGES_REQUESTED** [review]
@robodoo r-

image

Here I am website editor & designers and no other rights (I basically switched the user to portal user to lose all rights and then to internal + added website right).

File: addons/website_event/views/website_visitor_views.xml
Comment (bso-odoo) [reply addons/website_event/views/website_visitor_views.xml:38]: There was a similar issue on `website_event_track`.

**Comment by bso-odoo** [comment]
@rdeodoo Seems ok on runbot for 14.0 - no additional undetected failure on that form. Will check during forwards for additional `invisible` conditions.

File: addons/website_event_track/views/website_visitor_views.xml
Comment (rdeodoo) [inline addons/website_event_track/views/website_visitor_views.xml:24]: https://github.com/odoo/odoo/blob/14.0/addons/website_event_track/security/ir.model.access.csv/#L15-L16

Are you sure about this? `group_event_user` has acces to that? Shouldn't it be `group_event_manager`

**Review by rdeodoo — CHANGES_REQUESTED** [review]

File: addons/website_event_track/views/website_visitor_views.xml
Comment (bso-odoo) [reply addons/website_event_track/views/website_visitor_views.xml]: Indeed, I was mislead by `event_track_wishlisted_count` being `groups='event.group_event_user'`.
<https://github.com/odoo/odoo/blob/14.0/addons/website_event_track/models/website_visitor.py#L22>

**Review by rdeodoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
